x-pack/plugins/actions/server/builtin_action_types/alert_history_es_index/create_alert_history_index_template.ts

```diff
@@ -0,0 +1,100 @@
+/*
```

Comment:
This is similar enough to event_log, it made me think about the ILM policy it creates.  Do we need one here?  Or is the thinking you don't have this on all the time?  Or it's easy enough to "ILM" an index after it's been created (pretty sure that is not true!).

Which then makes me think - isn't this what [Data streams](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html) were built for?  Did this already come up and I'm forgetting? Re-reading that, I thought some of the complexity of the index templates, alias, ILM policy were reduced when creating a data stream, but ... still seems like we need to manually create most of that.

Perhaps there's a silver lining here: https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html#convert-index-alias-to-data-stream

customer would:

- create an alias for the index
- pre-reqs to convert an alias to a datastream, which would require some more work from the user, but not clear how much more than just creating the alias
- call the `POST _data_stream/_migrate/alert-history-blorg` API to do the conversion

Reply:
I initially tried to create an ILM policy, similar to how the event log bootstraps, but I found you need to create that initial index and link it to the alias used by the ILM (which event log does) but we can't do here because we're doing it at plugin start as the kibana_system user which can't write to indices that are not `.kibana` prepended.

I am not familiar with data streams at all...I'll do a little reading up on them.

Reply:
@pmuellr I tried out using data streams and the process was pretty seamless with adding an ILM policy. However, when testing out the rollover, I get a `security_exception`

```
security_exception: action [indices:admin/rollover] is unauthorized for user [kibana_system] with roles [kibana_system] on indices [.ds-alert-history-alway-firing-2021.04.01-000001,alert-history-alway-firing], this action is granted by the index privileges [manage_follow_index,manage,all]
```

so it looks like the user might have to do some extra work to get the rollover working, which is not ideal. This would probably show up regardless of if we use data streams or not if we apply an ILM policy since it looks like the kibana system user has default `view_index_metadata` and `monitor` on `*` indices and only `all` on `.kibana*` indices.

Reply:
ah ... the totally expected unknown unknowns!

Then, rather than try to figure that out, I think what we would want to do short term is just what you have now, with a thought to potentially extend this in the future to use ILM directly or data streams; presumably as some other configuration option, etc.  

So for now I think we'd just add some guidance in the doc that this index could grow and will need to be maintained manually by the customer.  Not sure if it would be appropriate to paste a delete_by_query example of deleting docs > than some time range, or at least suggesting (with a link) that manual delete by query would be one way to do this maintenance.